src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -477,12 +489,27 @@ private void Initialize(AnalyzerExecutor analyzerExecutor, DiagnosticQueue diagn
             };
 
             var analyzerExecutor = AnalyzerExecutor.Create(
-                compilation, analysisOptions.Options ?? AnalyzerOptions.Empty, addNotCategorizedDiagnostic, newOnAnalyzerException, analysisOptions.AnalyzerExceptionFilter,
-                IsCompilerAnalyzer, AnalyzerManager, ShouldSkipAnalysisOnGeneratedCode, ShouldSuppressGeneratedCodeDiagnostic, IsGeneratedOrHiddenCodeLocation, IsAnalyzerSuppressedForTree, GetAnalyzerGate,
+                compilation,
+                analysisOptions.Options ?? AnalyzerOptions.Empty,
+                addNotCategorizedDiagnostic,
+                newOnAnalyzerException,
+                analysisOptions.AnalyzerExceptionFilter,
+                IsCompilerAnalyzer,
+                AnalyzerManager,
+                ShouldSkipAnalysisOnGeneratedCode,
+                ShouldSuppressGeneratedCodeDiagnostic,
+                IsGeneratedOrHiddenCodeLocation,
+                IsAnalyzerSuppressedForTree,
+                GetAnalyzerGate,
                 getSemanticModel: GetOrCreateSemanticModel,
-                analysisOptions.LogAnalyzerExecutionTime, addCategorizedLocalDiagnostic, addCategorizedNonLocalDiagnostic, s => _programmaticSuppressions!.Add(s), cancellationToken);
+                addAdditionalFile: addAdditionalFile,
+                analysisOptions.LogAnalyzerExecutionTime,
+                addCategorizedLocalDiagnostic,
+                addCategorizedNonLocalDiagnostic,
+                s => _programmaticSuppressions!.Add(s),
+                cancellationToken);
```

Comment:
you guys have sooo many arguments passed to things.  it's a really strange design IMO, but i kept with it.